File: 575. Distribute Candies.py
class Solution(object):
    def distributeCandies(self, candies):
        """
        :type candies: List[int]
        :rtype: int
        """
        s = set(candies)
        if len(s) >= len(candies)/2:
            return len(candies)/2
        else:
            return len(s)

File: 575. Distribute Candies.go
func distributeCandies(candies []int)int{
	candiesMap := map[int]int{}
	// candiesMap := make(map[int]int)
	for _,i := range cnadies{
		if _,ok := candiesMap[i];ok{
			candiesMap[i]++
		}else{
			candiesMap[i] = 1
		}
	}
	if len(candiesMap) <= len(candies)/2{
		return len(candiesMap)
	}else{
		return len(candies)/2
	}
}